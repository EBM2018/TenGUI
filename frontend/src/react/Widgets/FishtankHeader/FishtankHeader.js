import React from 'react';

import PropTypes from 'prop-types';

export default class FishtankHeader extends React.PureComponent {
    static propTypes = {
      subject: PropTypes.string.isRequired,
      date: PropTypes.string.isRequired,
      my: PropTypes.string,
      chapitre: PropTypes.string.isRequired,
    };

    static defaultProps = {
      my: '',
    };

    render() {
      const {
        my, subject, date, chapitre,
      } = this.props;
      return (
        <div className="columns add-margin">
          <div className="column is-6">
            <h1 className="title is-2">
              {my}
              {' '}
                    Aquarium :
              {subject}
            </h1>
            <p className="is-4">
                  SÃ©ance ouverte le
              {' '}
              {date}
            </p>
            <p className="is-4">
                  Chapitre
              {': '}
              {chapitre}
            </p>
          </div>
        </div>
      );
    }
}
