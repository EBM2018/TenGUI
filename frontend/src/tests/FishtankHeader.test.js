import FishtankHeader from '../react/Widgets/FishtankHeader/FishtankHeader';
import React from 'react';
import renderer from 'react-test-renderer';


test ('derp',()=>{
    expect('True').toBeTruthy;
});
test('does this render', ()=> {
    const component=renderer
        .create(<FishtankHeader date={"12/03/2019"} subject={"Testing in JS"}/>);
    let tree=component.toJSON();
    expect(tree).toMatchSnapshot();
});
