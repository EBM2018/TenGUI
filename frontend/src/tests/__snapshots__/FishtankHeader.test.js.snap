// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does this render 1`] = `
<div
  id="NameHeader"
>
  <div>
     
    Aquarium :
    Testing in JS
  </div>
  <div>
    SÃ©ance ouverte le
     
    12/03/2019
  </div>
</div>
`;
